Added MSFT_EXOOMEConfiguration + MSFT_AADNamedLocationPolicy fix

William-Francillette opened the pull request:
<!--
    Thanks for submitting a Pull Request (PR) to this project.
    Your contribution to this project is greatly appreciated!

    Please make sure you have read the [Contribution Guidelines](https://github.com/PowerShell/SharePointDsc/wiki/Contributing%20to%20SharePointDsc).

    Please prefix the PR title with the resource name,
    e.g. 'ResourceName: My short description'.
    If this is a breaking change, then also prefix the PR title
    with 'BREAKING CHANGE:',
    e.g. 'BREAKING CHANGE: ResourceName: My short description'.

    You may remove this comment block, and the other comment blocks, but please
    keep the headers and the task list.
-->
#### Pull Request (PR) description
<!--
    Replace this comment block with a description of your PR.
-->

#### This Pull Request (PR) fixes the following issues
<!--
    If this PR does not fix an open issue, replace this comment block with None.
    If this PR resolves one or more open issues, replace this comment block with
    a list the issues using a GitHub closing keyword, e.g.:
    - Fixes #123
    - Fixes #124
-->


Hi, 
Find EXOOMEConfiguration module added to this PR and also a fix for AADNamedLocationPolicy when the policy is to be written to a tenant where it doesn't exist previously - If the -ErrorAction is not set, the code execution will stop from the try and then not create the policy

I hope this will help 
regards,
William

NikCharlebois commented on Modules/Microsoft365DSC/DSCResources/MSFT_AADNamedLocationPolicy/MSFT_AADNamedLocationPolicy.psm1:
```diff
@@ -411,6 +413,7 @@ function Export-TargetResource
     )
     $ConnectionMode = New-M365DSCConnection -Workload 'MicrosoftGraph' -InboundParameters $PSBoundParameters
 
+    Select-MgProfile -Name 'Beta'
```
Why are we forcing beta? The Get-MgIdentityConditionalAccessNamedLocation is available on the v1.0 endpoints. We should always try to prioritize using v1.0 over beta when available.

William-Francillette replied:
Updated

NikCharlebois commented on Modules/Microsoft365DSC/DSCResources/MSFT_MEMEndpointProtectionApplicationControlPolicyWindows10/MSFT_MEMEndpointProtectionApplicationControlPolicyWindows10.psm1:
```diff
@@ -0,0 +1,495 @@
+        $CertificateThumbprint
+    )
+
+    Write-Verbose -Message "Checking for the MEM Endpoint Protection Application Control Policy {$DisplayName}"
```
Also update verbose message to refer to Intune. This will be reviewed across all resources if a rebranding becomes necessary in the future.

William-Francillette replied:
Updated

NikCharlebois commented on Modules/Microsoft365DSC/DSCResources/MSFT_MEMEndpointProtectionApplicationControlPolicyWindows10/readme.md:
```diff
@@ -0,0 +1,4 @@
+
+# MEMEndpointProtectionApplicationControlPolicyWindows10
```
Update name and description to reflect renaming

William-Francillette replied:
Updated

NikCharlebois commented on Modules/Microsoft365DSC/DSCResources/MSFT_MEMEndpointProtectionApplicationControlPolicyWindows10/settings.json:
```diff
@@ -0,0 +1,18 @@
+{
+    "resourceName": "MEMEndpointProtectionApplicationControlPolicyWindows10",
```
Update name

William-Francillette replied:
Updated

NikCharlebois reviewed (changes requested):
Thanks for this. A few items to review. Also, for the Intune resource's schema.mof, make sure to specify ValueMap when available. Eg. the AppLockerApplication control one would need it. Thanks

William-Francillette commented on Modules/Microsoft365DSC/DSCResources/MSFT_MEMEndpointProtectionApplicationControlPolicyWindows10/MSFT_MEMEndpointProtectionApplicationControlPolicyWindows10.psm1:
```diff
@@ -0,0 +1,495 @@
+function Get-TargetResource
```
Updated

William-Francillette commented:
Hey,

I have updated AADNamedLocationsPolicy, and IntuneApplicationControlPolicyWindows10

I also added EXOPerimeterConfig and EXOResourceConfig and modified EXOOMEConfiguration to make sure the exception is raised when an error occurs by leveraging the error stream output because those cmdlet don't include ErrorAction attribute

I hope this helps!
Will

NikCharlebois reviewed (approved):
Great work, thank you so much for your contribution to the Microsoft365DSC project!
